numpy/add_newdocs.py

```diff
@@ -463,6 +463,50 @@ def luf(lamdaexpr, *args, **kwargs):
 
     """))
 
+add_newdoc('numpy.core', 'nested_iters',
+    """
+    Create a outermost-first tuple of iterators. They are nested so that
+    advancing one will change the inner iterators to point at the new
+    element in the outer iterator.
+
+    Parameters 
+    ----------
+    op : ndarray or sequence of array_like
+        The array(s) to iterate over.
+
+    axes : list of list of int
+        Each item is used as an "op_axes" argument to an nditer
+
+    flags, op_flags, op_dtypes, order, casting, buffersize (optional)
+        See `nditer` parameters of the same name
+
+    Returns
+    -------
+    iters : tuple of nditer
+        An nditer for each item in `axes`, outermost first
+
+    See Also
+    --------
+    nditer
+
+    Examples
+    --------
+
+    Basic usage. Note how vals is a "flattened" version of 
+    [a[:, 0, :], a[:, 1, 0], a[:, 2, :]] since we specified
+    that the first iter's axes as [1]
+
+    >>> a = np.arange(12).reshape(2, 3, 2)
+    >>> i, j = np.nested_iters([a, a*10], [[1], [0, 2]])
+    >>> vals = []
+    >>> for x in i:
+    ...    vals.append(map(int, y) for y in j])
+    >>> vals
+    [[0, 1, 6,  7],
+     [2, 3, 8,  9],
+     [4, 5, 10, 11]] 
```

Comment:
Where did `a * 10` go?

Reply:
arghh, didn't want to commit that, the printed output is way too confusing